core/io/resource_uid_database.cpp

```diff
@@ -0,0 +1,284 @@
+	if (path.is_empty()) {
+		return "unknown";
+	}
+
+	return "asset";
+}
+
+void UIDDB::ensure_db_dir() {
+	String dir = db_path.get_base_dir();
+	Ref<DirAccess> da = DirAccess::open(dir);
+	if (da.is_valid()) {
+		da->make_dir_recursive(dir);
+	}
+
+	ConfigFile cfg;
+	if (cfg.load(db_path) != OK) {
+		return;
+	}
+	const String section = "UIDs";
+	if (!cfg.has_section(section)) {
+		return;
+	}
+	PackedStringArray keys = cfg.get_section_keys(section);
+	for (int i = 0; i < keys.size(); ++i) {
+		String k = keys[i];
+		if (k.is_empty()) {
+			continue;
+		}
+		entries.insert(k, cfg.get_value(section, k));
+	}
+}
+
+void UIDDB::save_db() {
+	MutexLock ml(mutex);
+
+	// sort by created_on_ts oldest → newest
+	Vector<std::pair<int64_t, String>> order;
+	order.reserve(entries.size());
+
+	for (const KeyValue<String, Variant> &e : entries) {
+		if (e.key.is_empty()) {
+			continue;
+		}
+
+		int64_t ts = 0;
+		if (e.value.get_type() == Variant::DICTIONARY) {
+			ts = ((Dictionary)e.value).get("created_on_ts", (int64_t)0);
+		}
+		order.append({ ts, e.key });
+	}
+	order.sort();
+
+	Ref<FileAccess> f = FileAccess::open(db_path, FileAccess::WRITE);
+	if (f.is_valid()) {
+		f->store_line(";This file is automatically generated by the engine. it stores all created UIDs of Scripts, Scenes, Resources and other Assets ever generated by engine.");
+		f->store_line("");
+		f->store_line("[UIDs]");
+		f->store_line("");
+
+		for (int i = 0; i < order.size(); ++i) {
+			const String &key = order[i].second;
+			Variant val = entries[key];
+
+			Dictionary dict;
+			if (val.get_type() == Variant::DICTIONARY) {
+				Dictionary src = val;
+				dict["type"] = src.get("type", String());
+				dict["filename"] = src.get("filename", String());
+				dict["path"] = src.get("path", String());
+				dict["created_on"] = src.get("created_on", String());
+				dict["created_on_ts"] = src.get("created_on_ts", (int64_t)0);
+				dict["uid_u64"] = src.get("uid_u64", (uint64_t)0);
+
+				for (const Variant &k : src.keys()) {
+					String sk = k;
+					if (!dict.has(sk)) {
+						dict[sk] = src[sk];
+					}
+				}
+			} else {
+				dict = val;
+			}
+
+			f->store_line("\"" + key + "\" = {");
+
+			// Write fields
+			Array dkeys = dict.keys();
+			for (int j = 0; j < dkeys.size(); ++j) {
+				String k = dkeys[j];
+				Variant v = dict[k];
+
+				String line = "\t\"" + k + "\": ";
+				if (v.get_type() == Variant::STRING) {
+					line += "\"" + String(v).c_escape() + "\"";
+				} else {
+					line += String(v);
+				}
+				if (j < dkeys.size() - 1) {
+					line += ",";
+				}
+				f->store_line(line);
+			}
+
+			if (i < order.size() - 1) {
+				f->store_line("},");
+				f->store_line("");
+			} else {
+				f->store_line("}");
+			}
+		}
+
+		if (f->get_error() == OK) {
+			f->close();
+			return;
+		}
+		f->close();
+	}
+
+	// Fallback: if save failed, use ConfigFile (less pretty formatting).
+	ConfigFile cfg;
+	const String section = "UIDs";
+
+	for (const auto &p : order) {
```

Comment:
`auto` is [not allowed](https://contributing.godotengine.org/en/latest/engine/guidelines/cpp_usage_guidelines.html#auto-keyword)